server/opendp_apps/analysis/testing/test_dp_histogram_spec.py

```diff
@@ -0,0 +1,387 @@
+                      }
+
+        dp_hist = DPHistogramSpec(spec_props)
+        self.assertTrue(dp_hist.is_chain_valid())
+
+        bad_impute_info = [(-10, astatic.ERR_IMPUTE_PHRASE_MIN)]
+
+        for bad_impute, stat_err_msg in bad_impute_info:
+            print(f'> bad impute: {bad_impute}')
+            new_props = spec_props.copy()
+            new_props['fixed_value'] = bad_impute
+            dp_hist2 = DPHistogramSpec(new_props)
+
+            self.assertFalse(dp_hist2.is_chain_valid())
+            err_dict = dp_hist2.get_error_msg_dict()
+            print(f"  - {err_dict['message']}")
+            self.assertTrue(err_dict['message'].find(stat_err_msg) > -1)
+
+        good_impute_info = [-8, 5, '-8', '5', -7, 0, '0']
+
+        for good_impute in good_impute_info:
+            print(f'> good impute: {good_impute}')
+            new_props = spec_props.copy()
+            new_props['fixed_value'] = good_impute
+            dp_hist = DPHistogramSpec(new_props)
+            self.assertTrue(dp_hist.is_chain_valid())
+
+    # @skip
+    def test_100_run_dphist_calculation(self):
+        """(100) Run DP mean calculation"""
+        msgt(self.test_100_run_dphist_calculation.__doc__)
+
+        spec_props = {'variable': 'EyeHeight',
+                      'col_index': 19,
+                      'statistic': astatic.DP_HISTOGRAM,
+                      'dataset_size': 183,
+                      'epsilon': 1.0,
+                      'delta': 0.0,
+                      'ci': astatic.CI_95,
+                      # 'accuracy': None,
+                      'missing_values_handling': astatic.MISSING_VAL_INSERT_FIXED,
+                      'fixed_value': 5,
+                      'variable_info': {'min': -8,
+                                        'max': 5,
+                                        'type': 'Float', },
+                      }
+
+        dp_hist = DPHistogramSpec(spec_props)
+        print(dp_hist.has_error(), dp_hist.error_messages)
+        self.assertTrue(dp_hist.is_chain_valid())
+        if dp_hist.has_error():
+            print(dp_hist.get_error_messages())
+            return
+        # print('\nUI info:', json.dumps(dp_hist.get_success_msg_dict()))
+
+        # ------------------------------------------------------
+        # Run the actual mean
+        # ------------------------------------------------------
+        # Column indexes - We know this data has 20 columns
+        col_indexes = [idx for idx in range(0, 20)]
+
+        # File object
+        #
+        eye_fatigue_filepath = join(TEST_DATA_DIR, 'Fatigue_data.tab')
+        # print('eye_fatigue_filepath', eye_fatigue_filepath)
+        self.assertTrue(isfile(eye_fatigue_filepath))
+
+        file_obj = open(eye_fatigue_filepath, 'r')
+
+        # Call run_chain
+        #
+        dp_hist.run_chain(col_indexes, file_obj, sep_char="\t")
+
+
+
+    def test_105_run_dphist_calculation_categorical(self):
```

Comment:
Is this result correct? 
```
DP Histogram: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 184]
```

Reply:
No, I've fixed the categories so that it actually counts the values properly